### DISCUSSION TITLE
UDP input and RTL receiving in same time

### DISCUSSION THREAD

**Discussion by Taxom** [jvde-github/AIS-catcher#94] [Q&A] [answered]
In general, I have a question whether the simultaneous receiving and display of data from the UDP port (other AIS-cather) and receiver (SDR) is possible?

**Comment by Taxom** [comment, marked as answer]
Yep it works! But doesn't show plugin receiver on the second ais-catcher.

![Screenshot 2023-03-20 at 11-23-42 AIS-catcher MaxPi_AIS_Toronto](https://user-images.githubusercontent.com/20313019/226389310-0232ae77-2aa8-411b-8dc0-9010ddaf79eb.png)

![Screenshot 2023-03-20 at 11-23-11 AIS-catcher MaxPi_AIS_Toronto](https://user-images.githubusercontent.com/20313019/226389400-cbf56be4-ea67-4074-b7b8-0f887d1f346d.png)

**Comment by jvde-github** [comment]
Thanks for testing. I created a commit that shows all receivers in the summary page to avoid this confusion going forward.

How does the second receiver pick up the C&D channel? What is behind it?

**Reply by Taxom** [reply]
I use the old OrangePi Zero with the HackRF One to receive C/D channel. This test assembling with the indoor aerial. 
Can you add possibility to see in the table from which channel the messages was received?
for example:
![Screenshot 2023-03-21 at 12-37-00 AIS-catcher MaxPi_AIS_Toronto](https://user-images.githubusercontent.com/20313019/226683295-68d251be-2bbe-40eb-8dd5-cc3628a6221a.png)

**Reply by Taxom** [reply]
Thanks! It works.
The sampling rate shows only for UDP 
![Screenshot 2023-03-21 at 13-15-25 AIS-catcher MaxPi_AIS_Toronto](https://user-images.githubusercontent.com/20313019/226689874-20522a0d-7599-47f1-8a16-7640301b9e84.png)

**Reply by jvde-github** [reply]
I added the sample rates as well. Thanks

**Comment by jvde-github** [comment]
Worthwhile to consider indeed, thanks. I had the same issue looking through the list.

**Reply by jvde-github** [reply]
will also fix the sampling rate sometime. You could use HackRF to receive both AB and CD in parallel with the experimental branche btw.

**Comment by jvde-github** [comment]
![image](https://user-images.githubusercontent.com/52420030/226743845-b2c26fa3-59d6-410d-87c6-1b2122c937ba.png)

**Reply by Taxom** [reply]
Thanks! 
![Screenshot 2023-03-21 at 18-03-40 AIS-catcher](https://user-images.githubusercontent.com/20313019/226751954-691848ad-96b0-491d-8c51-bdf9f4c00e1d.png)

**Comment by otm64** [comment]
I use the -c CD AB option as it is the only way I have found for MarineTraffic to process the type 27 messages I forward. On my local ships table this means everything shows up as either channels A or B. Is there a creative way to code this so the local table expresses the original channel of C/D?

**Reply by jvde-github** [reply]
What we could do is to use the AB in -c CD AB to set the channels as used for NMEA but not how we use it internally in the program. Needs a small patch. Ideally you set it per UDP output but that requires quite some rework.
